Jieun/services/userService.js

```diff
@@ -1,28 +1,38 @@
-const userDao = require("../models/userDao");
 const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const userDao = require("../models/userDao");
+const {
+  passwordValidationCheck,
+  emailValidationCheck,
+} = require("../utils/validation-check");
 
 const signUp = async (name, profileImage, email, password, phoneNumber) => {
-  const pwValidation = new RegExp(
-    "^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})"
-  );
-  if (!pwValidation.test(password)) {
-    const err = new Error("PASSWORD_IS_NOT_VALID");
-    err.statusCode = 409;
-    throw err;
-  }
+  await passwordValidationCheck(password);
+  await emailValidationCheck(email);
+
   const saltRounds = 10;
   const hashedPassword = await bcrypt.hash(password, saltRounds);
-  const createUser = await userDao.createUser(
+  return userDao.createUser(
     name,
     profileImage,
     email,
     hashedPassword,
     phoneNumber
   );
+};
+
+const login = async (email, password) => {
+  const [user] = await userDao.getUserByEmail(email);
+  if (!user || !bcrypt.compare(password, user.password))
+    throw new Error("Invalid Email or Password");
```

Comment:
- user 정보를 Dao에서 가져올 때부터 객체로 가져오세요. 그럴려면 Dao를 수정 해야겠죠?
- if가 시작 될 때는 공백을 주어 가독성을 높여 주세요.
```suggestion
  const user = await userDao.getUserByEmail(email);
  
  if (!user || !bcrypt.compare(password, user.password))
    throw new Error("Invalid Email or Password");
```